scripts/zephyr_module.py

```diff
@@ -447,9 +430,91 @@ def git_revision(path):
                 return revision, False
         return None, False
 
-    zephyr_revision, zephyr_dirty = git_revision(zephyr_base)
-    zephyr_project = {'path': zephyr_base,
-                      'revision': zephyr_revision}
+    def git_remote(path):
+        popen = subprocess.Popen(['git', 'remote'],
+                                 stdout=subprocess.PIPE,
+                                 stderr=subprocess.PIPE,
+                                 cwd=path)
+        stdout, stderr = popen.communicate()
+        stdout = stdout.decode('utf-8')
+
+        remotes_name = []
+        if not (popen.returncode or stderr):
+            remotes_name = stdout.rstrip().split('\n')
+
+        remote_url = None
+        if len(remotes_name) == 1:
```

Comment:
> so http:// or https:// without git, doesn't seem to be sufficient.

The URL is transformed in ```writer.py``` because it's related to SPDX.
Just like you said that `-` should be replaced in ```writer.py``` and not in ```walker.py```.

> Note, a Zephyr module doesn't have to be a west project, also it doesn't even need to be a git repo.
So those cases must also be properly handled.

I agree, but before my PR, only git repository was supported. The only way to get the revision was through the ```git_revision``` function. So even if I agree with you, I would prefer to keep the current state of support and extend it later if possible.

I will have a look at how to get the same data returned by ```west list```, however I would need to know how I should handle the case where one or more remotes are added manually to a module.
It echoes a previous comment but:
What if a repository is checked out to a commit presents in a different remote than the one provided by ```west list```. What if the commit is in multiple remotes (and not necessarily in the one listed in ```west list```) ? Which remote should I choose ?

Reply:
> What if a repository is checked out to a commit presents in a different remote than the one provided by west list.

That is why the build output meta appends `-off` to the commit when the sha used is different from the sha defined in the manifest.
https://docs.zephyrproject.org/latest/kconfig.html#CONFIG_BUILD_OUTPUT_META

Even with your proposal of relying on a single remote will provide wrong information in case a user does a local commit, because in such a case the sha is not available at the remote.

Also users can do a `git fetch <url>` followed by a `git checkout FETCH_HEAD`.
In such a case there may be a single remote, but a commit from different url is checked out locally.
Such workflow is very common for testing PRs or branches.

> What if the commit is in multiple remotes (and not necessarily in the one listed in west list) ? Which remote should I choose ?

When `west` is used, then use the `west remote` if the sha is identical to the sha defined by the current manifest.
Because in that case we know the sha can be reached by that remote.
If it's not identical, then consider the sha `-off`. No reason to re-invent what `CONFIG_BUILD_OUTPUT_META` already provides. (Have you looked at this feature at all ?)

Reply:
I will use the `-off`. Thanks.
So, if I understand correctly, even though there is multiple remotes, I will use the remote returned by `west list`.

Reply:
@tejlmand I don't think there is any remote provided by west for the manifest repository.

If running `west init -m https://github.com/zephyrproject-rtos/zephyr.git`, `west list` does not provide any remote for the manifest. I didn't find anything in west code to have this information (command or function), you can only extract the manifest directory. The only solution I have is to run `git remote` in the manifest directory to get it.

So if I have two remotes (for example, zephyr remotes and the forked zephyr to add my feature), I don't have any clue on which remotes I should use.

What should I do in this case ? No remote ? The first one ?

Thanks.